lib/src/loader.dart

```diff
@@ -27,7 +27,7 @@ class Loader {
 
   /// The span of the entire stream emitted so far.
   FileSpan get span => _span;
-  FileSpan _span;
+  late FileSpan _span;
```

Comment:
This does not block the PR - opening here for a quick discussion.

it is possible to make this not be `late`, but it's a little convoluted.

cc @lrhn @jakemac53 I'm curious what you think of this pattern.

(This does make me wish again for more expressive initializer lists, like being able to refer to previously assigned fields in later ones in initializer lists... https://github.com/dart-lang/language/issues/622)

```diff
diff --git a/lib/src/loader.dart b/lib/src/loader.dart
index 3705406..1b0a846 100644
--- a/lib/src/loader.dart
+++ b/lib/src/loader.dart
@@ -27,17 +27,19 @@ class Loader {

   /// The span of the entire stream emitted so far.
   FileSpan get span => _span;
-  late FileSpan _span;
+  FileSpan _span;

   /// Creates a loader that loads [source].
   ///
   /// [sourceUrl] can be a String or a [Uri].
   Loader(String source, {sourceUrl})
-      : _parser = Parser(source, sourceUrl: sourceUrl) {
-    var event = _parser.parse();
-    _span = event.span;
-    assert(event.type == EventType.streamStart);
-  }
+      : this._fromParser(Parser(source, sourceUrl: sourceUrl));
+
+  Loader._fromParser(Parser parser) : this._fromEvent(parser, parser.parse());
+
+  Loader._fromEvent(this._parser, Event event)
+      : _span = event.span,
+        assert(event.type == EventType.streamStart);

   /// Loads the next document from the stream.
   ///
```

We could also make it simpler with a factory constructor, but that is technically another breaking change and I haven't looked at the impact quite yet.

Reply:
Ah, this class is not in the public API, so it's not a breaking change to make it a `factory`.

https://github.com/dart-lang/yaml/pull/96

Reply:
LGTM.  Thanks!

Reply:
The redirecting generative constructor approach is how I would solve the problem if it was *necessary*.
If it can be rewritten to be a factory, then that's obviously simpler.